scripts/update-component-sha.sh

```diff
@@ -1,10 +1,66 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# can be run in 2 modes:
+# 1- $0 --hash $oldhash $newhash              - replace all instances of $oldhash with $newhash
+# 2- $0 --image $image $newhash        - replace all hashes for $image with $newhash
+#
+# for backwards compatibility if no argtype is given, the first method is assumed. Thus the following are equivalent:
+#   $0 $oldhash $newhash 
+#   $0 --hash $oldhash $newhash 
+#
+
+function usage() {
+    echo "Modes of operation:" >&2
+    echo >&2
+    echo "Replace by hash:" >&2
+    echo "	$0 --hash <OLD> <NEW>" >&2
+    echo "	Example: $0 8675309abcdefg abcdef567899" >&2
```

Comment:
I suppose in a way your usage matches the `--hash` name you've given it, whereas mine might better be called `--tag`, but given the implementation would be identical (and actually you can omit the argument anyway) then I don't think we need to have both...